platform/lib/build/pageMinifier.js

```diff
@@ -87,22 +90,39 @@ class PageMinifier {
   start(path) {
     // Ugly but needed to keep scope for .pipe
     const scope = this;
-
     return gulp.src(`${path}/**/*.html`, {'base': './'})
-        .pipe(through.obj(function(page, encoding, callback) {
-          scope._log.await(`Minifying ${page.relative} ...`);
+        .pipe(through.obj(async function(canonicalPage, encoding, callback) {
+          let html = canonicalPage.contents.toString();
+          html = scope.minifyPage(html, canonicalPage.path);
+
+          scope._log.info('Optimizing ${canonicalPage.relative}');
+          const ampPath = canonicalPage.relative.replace('.html', '.amp.html');
+          const optimizedHtml = await scope.optimize(html, ampPath);
+
+          const ampPage = canonicalPage.clone();
+          ampPage.path = ampPath;
 
-          let html = page.contents.toString();
-          html = scope.minifyPage(html, page.path);
-          page.contents = Buffer.from(html);
+          canonicalPage.contents = Buffer.from(optimizedHtml);
+          ampPage.contents = Buffer.from(html);
 
-          this.push(page);
+          this.push(canonicalPage);
+          this.push(ampPage);
 
           callback();
         }))
         .pipe(gulp.dest('./'));
   }
 
+
+  async optimize(html, path) {
+    const ampRuntimeVersion = await runtimeVersionPromise;
+    return ampOptimizer.transformHtml(html, {
+      ampUrl: path.replace(/^pages/, ''),
```

Comment:
@matthiasrohmer I'm sure there's a better way to do this

Reply:
None that I can think of quickly... 🤔